src/raft_server.cxx

```diff
@@ -455,7 +455,18 @@ ptr<resp_msg> raft_server::process_req(req_msg& req) {
          req.get_type() == msg_type::request_vote_request ||
          req.get_type() == msg_type::install_snapshot_request ) {
         // we allow the server to be continue after term updated to save a round message
-        update_term(req.get_term());
+        bool term_updated = update_term(req.get_term());
+
+        if ( !im_learner_ &&
+             !hb_alive_ &&
+             term_updated &&
+             req.get_type() == msg_type::request_vote_request ) {
+            // If someone has newer term, that means leader has not been
+            // elected in the current term, and this node's election timer
```

Comment:
this is misleading, a newer term does not mean leader not elected in the current term, think about this, a follower did not receive hb for quite some time and initiate a new term, there could be a leader in previous term

Reply:
Thanks Andy. We have pre-vote so that follower cannot increase its term without the agreement of quorum. The increment of term only happens when more than majority of servers don't receive heartbeat from the (possibly previous) leader.